lib/ansible/cli/__init__.py

```diff
@@ -324,7 +324,16 @@ def ask_passwords():
         becomepass = None
         become_prompt = ''
 
-        become_prompt_method = "BECOME" if C.AGNOSTIC_BECOME_PROMPT else op['become_method'].upper()
+        if C.AGNOSTIC_BECOME_PROMPT:
+            become_prompt_method = "BECOME"
+        else:
+            display.deprecated(
+                'The setting AGNOSTIC_BECOME_PROMPT to False is deprecated and no longer used. '
+                'Ensure that you have configured any interactive script to accept "BECOME" '
+                'to identify become password prompt.',
+                version='2.22'
+            )
+            become_prompt_method = op['become_method'].upper()
 
         try:
             become_prompt = "%s password: " % become_prompt_method
```

Comment:
it defaults to true, the 'per become method prompt' is the one that should go away, also only show deprecated if the value does not come from 'default'

Reply:
https://ansible.readthedocs.io/projects/runner/en/stable/intro/#env-passwords